charts/cluster-api-cluster-openstack/templates/control-plane.yaml

```diff
@@ -73,23 +73,27 @@ spec:
   template:
     spec:
       flavor: {{ .Values.controlPlane.machine.flavor }}
-      image: {{ .Values.controlPlane.machine.image }}
+      image:
+        filter:
+          name: {{ .Values.controlPlane.machine.image }}
       {{- if .Values.openstack.sshKeyName }}
       sshKeyName: {{ .Values.openstack.sshKeyName }}
       {{- end }}
-      cloudName: {{ .Values.openstack.cloud }}
       identityRef:
         name: {{ include "cloudconfig.name" . }}
-        kind: Secret
+        cloudName: {{ .Values.openstack.cloud }}
       {{- if .Values.controlPlane.machine.disk }}
       rootVolume:
         availabilityZone: {{ .Values.openstack.volumeFailureDomain }}
-        diskSize: {{ .Values.controlPlane.machine.disk.size }}
+        sizeGiB: {{ .Values.controlPlane.machine.disk.size }}
       {{- end }}
       {{- with $cluster := .Values.cluster -}}
         {{- with $metadata := $cluster.serverMetadata -}}
           {{- printf "serverMetadata:" | nindent 6 }}
-          {{- toYaml $metadata | nindent 8 }}
+          {{ range $k, $v := $metadata -}}
```

Comment:
I _think_, I may be more comfortable with

```
          {{ range $k, $v := $metadata -}}
            {{- printf "- key: %s" $k | nindent 6 }}
            {{- printf "  value: %s" $v | nindent 6 }}
          {{ end -}}
```

as the nesting is less horrible.  Not that mustache isn't just horrible to start with.

Reply:
Done, this now looks like:

```
      serverMetadata:
      - key: cluster
        value: capitest
      - key: organization
        value: nscale
      - key: project
        value: dev
```